v1/README_Cloud_PubSub_to_Datadog.md

```diff
@@ -0,0 +1,199 @@
+Pub/Sub to Datadog Template
+---
+A pipeline that reads from a Pub/Sub subscription and writes to Datadog's Logs API.
+
+:memo: This is a Google-provided template! Please
+check [Provided templates documentation](https://cloud.google.com/dataflow/docs/guides/templates/provided/pubsub-to-datadog)
+on how to use it without having to build from sources using [Create job from template](https://console.cloud.google.com/dataflow/createjob?template=Cloud_PubSub_to_Datadog).
+
+
+:bulb: This is a generated documentation based
+on [Metadata Annotations](https://github.com/GoogleCloudPlatform/DataflowTemplates#metadata-annotations)
+. Do not change this file directly.
+
+## Parameters
+
+### Required Parameters
+
+* **inputSubscription** (Pub/Sub input subscription): Pub/Sub subscription to read the input from, in the format of 'projects/your-project-id/subscriptions/your-subscription-name' (Example: projects/your-project-id/subscriptions/your-subscription-name).
+* **site** (Datadog site.): Datadog site. This should be routable from the VPC in which the pipeline runs. See: https://docs.datadoghq.com/getting_started/site (Example: datadoghq.com).
+* **outputDeadletterTopic** (Output deadletter Pub/Sub topic): The Pub/Sub topic to publish deadletter records to. The name should be in the format of projects/your-project-id/topics/your-topic-name.
+
+### Optional Parameters
+
+* **apiKey** (Logs API key.): Datadog Logs API key. Must be provided if the apiKeySource is set to PLAINTEXT or KMS. See: https://docs.datadoghq.com/account_management/api-app-keys.
+* **batchCount** (Batch size for sending multiple events to Datadog Logs API.): Batch size for sending multiple events to Datadog Logs API. Defaults to 10. Max is 1000.
+* **parallelism** (Maximum number of parallel requests.): Maximum number of parallel requests. Default 1 (no parallelism).
+* **includePubsubMessage** (Include full Pub/Sub message in the payload.): Include full Pub/Sub message in the payload (true/false). Defaults to false (only data element is included in the payload).
```

Comment:
Originally removed this option, but added it back in realizing that our [existing procedure](https://docs.datadoghq.com/integrations/google_cloud_platform/#forward-logs-to-datadog) always sends the entire PubSub message as the log line. So, if a customer wants to replicate what already exists, they should set this to `true`. But we left the default as `false`, since it's highly unlikely a Datadog customer would want the entire PubSub message, since PubSub in this case is just a way to transfer logs.